### PR TITLE
HHH-15452 Predicate add expression results in UnsupportedOperationException

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-15452

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmBooleanExpressionPredicate.java [hunk 1]
```diff
@@ -73,7 +72,7 @@ public <T> T accept(SemanticQueryWalker<T> walker) {
 
 	@Override
 	public List<Expression<Boolean>> getExpressions() {
-		return Collections.singletonList( booleanExpression );
+		return Arrays.asList( booleanExpression );
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmBooleanExpressionPredicate.java:75]: The list which `Arrays.asList` returns can not be added to unfortunately, so we need to use this:

```suggestion
		final List<Expression<Boolean>> list = new ArrayList<>( 1 );
		list.add( booleanExpression );
		return list;
```

File: hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmGroupedPredicate.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ public boolean isNegated() {
 
 	@Override
 	public List<Expression<Boolean>> getExpressions() {
-		return Collections.singletonList( subPredicate );
+		return Arrays.asList( subPredicate );
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmGroupedPredicate.java:62]: The list which `Arrays.asList` returns can not be added to unfortunately, so we need to use this:

```suggestion
		final List<Expression<Boolean>> list = new ArrayList<>( 1 );
		list.add( subPredicate );
		return list;
```

File: hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmJunctionPredicate.java [hunk 3]
```diff
@@ -88,8 +86,9 @@ public boolean isNegated() {
 
 	@Override
 	public List<Expression<Boolean>> getExpressions() {
-		//noinspection unchecked
-		return (List<Expression<Boolean>>) (List<?>) predicates;
+		ArrayList<Expression<Boolean>> expressions = new ArrayList<>( predicates.size() );
+		expressions.addAll( predicates );
+		return expressions;
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmJunctionPredicate.java:91]:
```suggestion
		return new ArrayList<>( predicates );
```

File: hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmNegatedPredicate.java [hunk 4]
```diff
@@ -58,7 +58,7 @@ public SqmPredicate getWrappedPredicate() {
 
 	@Override
 	public List<Expression<Boolean>> getExpressions() {
-		return Collections.singletonList( wrappedPredicate );
+		return Arrays.asList( wrappedPredicate );
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmNegatedPredicate.java:61]: The list which `Arrays.asList` returns can not be added to unfortunately, so we need to use this:

```suggestion
		final List<Expression<Boolean>> list = new ArrayList<>( 1 );
		list.add( wrappedPredicate );
		return list;
```

File: hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmBooleanExpressionPredicate.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmBooleanExpressionPredicate.java]: Thanks @beikov

**Pull request merged**